src/target/riscv/riscv.c

```diff
@@ -655,20 +664,41 @@ static int set_trigger(struct target *target, unsigned int idx, riscv_reg_t tdat
 		return ERROR_FAIL;
 	if (riscv_reg_get(target, &tdata2_rb, GDB_REGNO_TDATA2) != ERROR_OK)
 		return ERROR_FAIL;
-	bool tdata1_config_denied = (tdata1 & ~tdata1_ignore_mask) != (tdata1_rb & ~tdata1_ignore_mask);
+	bool unsupported_napot_range = false;
+	bool tdata1_config_denied = true;
+	const uint32_t type = get_field(tdata1, CSR_TDATA1_TYPE(riscv_xlen(target)));
+	const bool is_mcontrol = type == CSR_TDATA1_TYPE_MCONTROL;
+	const bool is_napot_match = get_field(tdata1, CSR_MCONTROL_MATCH) == CSR_MCONTROL_MATCH_NAPOT;
+	if (is_mcontrol && is_napot_match) {
+		maskmax = get_field(tdata1_rb, CSR_MCONTROL_MASKMAX(riscv_xlen(target)));
+		const unsigned int napot_size = count_trailing_ones(tdata2) + 1;
+		if (maskmax < napot_size)
+			unsupported_napot_range = true;
+		tdata1_config_denied = (tdata1 & ~maskmax) != (tdata1_rb & ~maskmax);
+	} else if (is_mcontrol && !is_napot_match) {
+		maskmax = get_field(tdata1_rb, CSR_MCONTROL_MASKMAX(riscv_xlen(target)));
+		tdata1_config_denied = (tdata1 & ~maskmax) != (tdata1_rb & ~maskmax);
+	} else {
+		tdata1_config_denied = tdata1 != tdata1_rb;
+	}
 	bool tdata2_config_denied = tdata2 != tdata2_rb;
```

Comment:
Detail, but the use of `const` helps the code readers.

```suggestion
	const bool tdata2_config_denied = tdata2 != tdata2_rb;
```

Reply:
addressed